File: CompositeVisitor/CompositeVisitor_java/Rectangle.java
public class Rectangle implements IShape {
  
 private float m_w, m_h;
 
 public Rectangle(float a_w, float a_h) {
   m_w = a_w;
   m_h = a_h;
 }
  
  public float getArea() {
    return m_w * m_h;
  }
  
  public void accept(IShapeVisitor a_visitor) {
    a_visitor.visitRectangle(this);
  }
}

File: CompositeVisitor/CompositeVisitor_Swift_3.0.1/CompositeVisitor/model/Rectangle.swift
//
//  Rectangle.swift
//  CompositeVisitor
//
//  Created by Sunny Buttar on 2016-11-13.
//  Copyright © 2016 Sarpreet Singh. All rights reserved.
//

import Foundation

class Rectangle : IShape {
    
    private var m_width : Float;
    private var m_height : Float;
    
    init(a_width : Float, a_height : Float) {
        self.m_width = a_width;
        self.m_height = a_height;
    }
    
    func getArea() -> Float {
        return m_height * m_width;
    }
    
    func accept(a_visitor: IShapeVisitor) {
        a_visitor.visitRectangle(a_rect: self);
    }
}
